### PR TITLE
Breakout entities

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**

### CODE REVIEW

File: modelon/impact/client/entities/model_executable.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+from modelon.impact.client import exceptions
+
+
+logger = logging.getLogger(__name__)
+
+Options = Union[ExecutionOptions, Dict[str, Any]]
+
+
+def _assert_compilation_is_complete(status, operation_name="Operation"):
+    if status != ModelExecutableStatus.SUCCESSFUL:
```
Comment (efredriksson-modelon) [inline modelon/impact/client/entities/model_executable.py:22]: this line does not do anything

File: modelon/impact/client/entities/workspace.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+
+        Example::
+
+            workspace.create_experiment(definition)
+        """
+        if isinstance(
+            definition,
+            (SimpleFMUExperimentDefinition, SimpleModelicaExperimentDefinition),
+        ):
+            experiment_definition = definition.to_dict()
```
Comment (efredriksson-modelon) [inline modelon/impact/client/entities/workspace.py:484]: experiment_definition -> definition_as_dict would better describe what is happening here

File: modelon/impact/client/exceptions.py [hunk 3]
```diff
@@ -15,11 +15,20 @@ class OperationTimeOutError(Error):
 
 
 class OperationFailureError(Error):
-    pass
+    def for_operation(self, operation_name: str):
```
Comment (efredriksson-modelon) [inline modelon/impact/client/exceptions.py:18]: These methods should be static like:
```
 @classmethod
 def from_operation(cls, name: str):
    return cls(...) 
```

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
👍

File: modelon/impact/client/entities/experiment.py [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+from modelon.impact.client.operations import experiment
+from modelon.impact.client.entities.case import Case
+from modelon.impact.client.entities.asserts import assert_variable_in_result
+from modelon.impact.client import exceptions
+
+logger = logging.getLogger(__name__)
+
+
+def _assert_experiment_is_complete(status, operation_name="Operation"):
+    if status != ExperimentStatus.DONE:
```
Comment (efredriksson-modelon) [inline modelon/impact/client/entities/experiment.py:16]: same, does nothing

File: modelon/impact/client/entities/case.py [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+
+    SUCCESSFUL = "successful"
+    FAILED = "failed"
+    CANCELLED = "cancelled"
+    NOT_STARTED = "not_started"
+    STARTED = 'started'
+
+
+def _assert_case_is_complete(status, operation_name="Operation"):
+    if status == CaseStatus.SUCCESSFUL:
```
Comment (efredriksson-modelon) [inline modelon/impact/client/entities/case.py:33]: same, does nothing

**Comment by abhilash-kumar-nair** [comment]
@efredriksson-modelon , Thanks for the review! I have made the suggested fixes. Could you please take another look?

**Review by efredriksson-modelon — APPROVED** [review]
🍉

**Pull request merged**